internal/backend/db/dbgorm/triggers.go

```diff
@@ -29,6 +29,14 @@ func (gdb *gormdb) deleteTrigger(ctx context.Context, triggerID sdktypes.UUID) e
 		if err := tx.isCtxUserEntity(tx.ctx, triggerID); err != nil {
 			return err
 		}
+
+		// enforce foreign keys constrains while soft-deleting
+		var count int64
+		tx.db.Model(&scheme.Event{}).Where("deleted_at is NULL and trigger_id = ?", triggerID).Count(&count)
```

Comment:
not sure I get what you're doing here
looking for non deleted events ? 
is this a mix and match of soft delete and hard delete ?

Reply:
I also think you should add limit 1 to this count since you only care if something exists
otherwise you're going to count all relevant events for not real reason

Reply:
1. once you soft deleting you cannot enforce foreign keys in db, only in code. We are doing the same in env. Just to prevent deleting trigger before events reference it were deleted
2. regarding count and limit you are totally right

Reply:
ok, I understand now the delete part, so my first question is irrelevant

count should be fixed to something else

Reply:
I replaced count by select(1) limit(1)